File: threedollar-domain/src/main/java/com/depromeet/threedollar/domain/user/domain/review/ReviewCreator.java
package com.depromeet.threedollar.domain.user.domain.review;

import com.depromeet.threedollar.common.docs.ObjectMother;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@ObjectMother
@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class ReviewCreator {

    public static Review create(Long storeId, Long userId, String contents, int rating) {
        return Review.of(storeId, userId, contents, rating);
    }

    public static Review createDeleted(Long storeId, Long userId, String contents, int rating) {
        Review review = create(storeId, userId, contents, rating);
        review.delete();
        return review;
    }

}

File: threedollar-domain/src/testFixtures/java/com/depromeet/threedollar/domain/user/domain/review/ReviewCreator.kt
package com.depromeet.threedollar.domain.user.domain.review

import com.depromeet.threedollar.domain.user.domain.TestFixture

@TestFixture
object ReviewCreator {

    @JvmStatic
    fun create(
        storeId: Long,
        userId: Long,
        contents: String,
        rating: Int
    ): Review {
        return Review.builder()
            .storeId(storeId)
            .userId(userId)
            .contents(contents)
            .rating(rating)
            .build();
    }

    @JvmStatic
    fun createDeleted(
        storeId: Long,
        userId: Long,
        contents: String,
        rating: Int
    ): Review {
        val review = Review.builder()
            .storeId(storeId)
            .userId(userId)
            .contents(contents)
            .rating(rating)
            .build();
        review.delete()
        return review
    }

}
